.github/workflows/client.yml

```diff
@@ -0,0 +1,48 @@
+name: Client CI
+on:
+  push:
+    branches: main
+  pull_request:
+    branches: main
+jobs:
+  build-client:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+      - name: Npm install
+        run: npm install
```

Comment:
I believe you want `npm ci` rather than `npm install` here as well as everywhere else in these workflows.